Fixes attachments in post edit, ordering of image attachments

lorenjohnson opened the pull request:
Fixes the issue with images attachments not appearing in Post edit, and issues in both Post and Comment editors with retaining image ordering.

tibetsprague commented:
18:28:29 web.1      |      at Function.createForSubject (/Users/tibet/Dev/hylo/hylo-node/api/models/Media.js:73:29)

tibetsprague commented:
same with trying to add an image to a post

lorenjohnson commented:
The editing post images is now fixed in the attached PR and the updating issue was a bug in the refactoring on the backend branch which is now fixed and on the `dev` branch of `hylo-node`.
